src/utils/cluster.ts

```diff
@@ -85,6 +91,21 @@ export class ClusterUtils {
     return updateMask;
   }
 
+  static getClusterBaseConfigWithFullLocation(
```

Comment:
I think we should try to add a test that specifically asserts against this expected behaviour:

```
it('populates full location path if location parameter provided')
```

What format is it expected that metadata.location is in? Is this value provided by a user.

Reply:
I would be tempted to add a test for the user behavior that would cause these different code paths to be hit.

One test for  `location` being undefined, one test for `location` being set.